Samples/PowerShell/functions/func_ModernDashboards.ps1

```diff
@@ -4,6 +4,134 @@
+    PS C:\> Get-ModernDashboard -SwisConnection $SwisConnection -DashboardId 1 
+
+    PS C:\> Get-ModernDashboard -SwisConnection $SwisConnection -DashboardId 1 -IncludeSystem
+
+    The first command will (most likely) return nothing as DashboardId 1 is a "system" dashboard
+    The second command will return the details of Dashboard ID 1
+
+.NOTES
+    Author:  Kevin M. Sparenberg (https://thwack.solarwinds.com/members/kmsigma)
+    Version: 0.9
+    Last Updated: 2021-09-30
+    Validated: Orion Platform 2020.2.5
+
+    TBD List:
+        * Validate that the script works on non-Windows
+#>
+function Get-SwisModernDashboard {
+    [CmdletBinding(
+        DefaultParameterSetName = 'Normal', 
+        PositionalBinding = $false,
+        HelpUri = 'https://documentation.solarwinds.com/en/success_center/orionplatform/content/core-fusion-dashboard-import-export.htm',
+        ConfirmImpact = 'Low')]
+    [Alias('Get-ModernDashboard')]
+    [OutputType()]
+    Param
+    (
+        # The connection to the SolarWinds Information Service
+        [Parameter(
+            Mandatory = $false, 
+            ValueFromPipeline = $true,
+            ValueFromPipelineByPropertyName = $true, 
+            ValueFromRemainingArguments = $false)]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
+        [Alias("Swis")] 
+        [SolarWinds.InformationService.Contract2.InfoServiceProxy]$SwisConnection = $Global:SwisConnection,
+
+        # The dashboard Id we'll display
+        [AllowNull()]
+        [Alias("Id")] 
+        [int32[]]$DashboardId,
+
+        # Filter for specific owners?
+        [AllowNull()]
+        [string[]]$Owner,
+
+        # Should we include system Dashboards?
+        [AllowNull()]
+        [switch]$IncludeSystem,
+
+        # Omits white space and indented formatting in the output string.
+        [AllowNull()]
+        [switch]$IncludeJson
+
+    )
+
+    Begin {
+        # Define the Property List
+        $PropertyList = "DashboardID", "DisplayName", "UniqueKey", "Owner", "LastUpdate", "Private"
+
+        # Define default Query
+        $Swql = "SELECT DashboardID, ParentID, Owner, UniqueKey, LastUpdate, Private, IsSystem, DisplayName FROM Orion.Dashboards.Instances WHERE ParentID IS NULL"
+
+    }
+    Process {
+        
+        #region Filter for DashboardIds
+        # Add Dashboards if we are sent IDs
+        if ( $DashboardId ) {
+            # Add an = to the WHERE clause
+            $Swql += " AND DashboardID = $( $DashboardId )"
+        }
+        #endregion Filter for DashboardIds
+        
+        #region Filter for Owners
+        # If the Owner parameter is passed, we need to filter for that as well
+        # Same logic as above, but for the array, we need to be sure to include the single quotes as wrappers
+        if ( $Owner ) {
+            $Swql += " AND Owner = '$( $DashboardId )'"
```

Comment:
Why does this reference `$DashboardId` instead of `$Owner`?  Also, what if we get 2+ owners?  And this is subject to the SWQL equivalent of [SQL injection](https://owasp.org/www-community/attacks/SQL_Injection).  As a best practice, we should sanitize the owners before building a query via string concatenation.

Reply:
Nice catch - fixed - also did the `IN` thing as well.

Re: SWQL Injection, I may need to rethink how this is done as a whole.  The original intent was to be able to `Get-SwisModernDashboard -DashboardId 10 | Export-SwisModernDashboard -Path "C:\Export\Path\"`

I think I have another way to do this work with `ParameterSets` but it could get ugly.  How would you "sanitize" the language?  I'm assuming just check for keywords and throw an error on a match?

Reply:
I'm just going to comment out the whole thing because it needs a refactor.  I kept it in place so that I have the desired parameters I want to use.